plugin/kubernetes/handler.go

```diff
@@ -43,11 +43,13 @@ func (k Kubernetes) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.M
 	case dns.TypeSRV:
 		records, extra, err = plugin.SRV(ctx, &k, zone, state, plugin.Options{})
 	case dns.TypeSOA:
-		records, err = plugin.SOA(ctx, &k, zone, state, plugin.Options{})
+		if len(state.Name()) == len(zone) {
+			records, err = plugin.SOA(ctx, &k, zone, state, plugin.Options{})
+		}
 	case dns.TypeAXFR, dns.TypeIXFR:
 		return dns.RcodeRefused, nil
 	case dns.TypeNS:
-		if state.Name() == zone {
+		if len(state.Name()) == len(zone) {
```

Comment:
why does this need to be len? Same above

Reply:
This is done to make the comparison case insensitive.

Reply:
But it does not check the content, better to use `strings.EqualFold`

Reply:
The content is validated earlier.

Reply:
I think `if qname == zone {` would work too... I think the cases match there... Name() lowercases.

I think comparing len requires less operations, but perhaps thats's insignificant.

Reply:
If I may put my two cents in, slight improvement in performance is important, but if it does not harm the readability of the code. If two people have asked this question, then perhaps other developers will also ask similar questions.

Reply:
OK, doing string compare now.

I've also reverted the changes to the NS case, so it's doing the name compare as it was before. Fixing the case insensitivity/preservation for the NS case is off topic anyways.  It can be fixed in another PR.